### PR TITLE
Fix inconsistent final derivations in rigid monoidal categories

### LINKED ISSUE #929: Inconsistent final derivations of IsomorphismFromTensorProductToInternalHom and IsomorphismFromInternalHomToDual
The following final derivations are inconsistent: https://github.com/homalg-project/CAP_project/blob/8c610224cd4a24adb2e9f69b6bcd25e0eb00d5e5/MonoidalCategories/gap/RigidSymmetricClosedMonoidalCategoriesDerivedMethods.gi#L429-L538

The first two derivations set $\underline{Hom}( a, b ) = a^v \otimes b$ and thus in particular $\underline{Hom}( a, 1 ) = a^v \otimes 1$, while the last two derivations set $\underline{Hom}( a, 1 ) = a^v$. That is, the two interpretations differ by the right unitor. This gives the expected error in the following example:

```
LoadPackage( "LinearAlgebra" );
cat := MatrixCategory( HomalgFieldOfRationals( ) : no_precompiled_code );
LoadPackage( "LazyCategories" );
lazy := LazyCategory( cat : primitive_operations );
obj := 1 / cat / lazy;
PreCompose( IsomorphismFromTensorProductToInternalHom( obj, TensorUnit( lazy ) ), IsomorphismFromInternalHomToDual( obj ) );
```
gives
```
Error, in function PreCompose
       of category LazyCategory( Category of matrices over Q ):
       morphisms not composable
```

In addition to the missing right unitor, it would also be a good idea to use a single final derivation to install all four operations at once as in e.g. https://github.com/homalg-project/CAP_project/blob/master/MonoidalCategories/gap/HomomorphismStructureDerivedMethods.gi.

@TKuh Is this something you could have a look at?

### PR DESCRIPTION
**Pull request by TKuh**
Hopefully fixes #929.

I also added remarks about duals in the rigid case and documented the GAP property, which was somehow missing.

### CODE REVIEW

File: MonoidalCategories/gap/RigidSymmetricClosedMonoidalCategoriesDerivedMethods.gi [hunk 1]
```diff
@@ -450,89 +451,30 @@ AddFinalDerivation( IsomorphismFromTensorProductToInternalHom,
...
   function( cat, object )
     
-    return IdentityMorphism( cat, DualOnObjects( cat, object ) );
+    return IdentityMorphism( cat, TensorProductOnObjects( cat, DualOnObjects( cat, object ), TensorUnit( cat ) ) );
     
-end : CategoryFilter := IsRigidSymmetricClosedMonoidalCategory,
-      Description := "IsomorphismFromDualToInternalHom as the identity of the Dual" );
+  end
+] : CategoryFilter := IsRigidSymmetricClosedMonoidalCategory,
+    Description := "adding the isomorphisms between the internal hom and the tensor product as identities"
```
Comment (zickgraf) [inline MonoidalCategories/gap/RigidSymmetricClosedMonoidalCategoriesDerivedMethods.gi:479]:
```suggestion
    Description := "setting the isomorphisms between the internal hom and the tensor product with the dual object to the identities"
```

File: MonoidalCategories/gap/RigidSymmetricCoclosedMonoidalCategoriesDerivedMethods.gi [hunk 2]
```diff
@@ -448,89 +449,30 @@ AddFinalDerivation( IsomorphismFromTensorProductToInternalCoHom,
...
   function( cat, object )
     
-    return IdentityMorphism( cat, CoDualOnObjects( cat, object ) );
+    return IdentityMorphism( cat, TensorProductOnObjects( cat, TensorUnit( cat ), CoDualOnObjects( cat, object ) ) );
     
-end : CategoryFilter := IsRigidSymmetricCoclosedMonoidalCategory,
-      Description := "IsomorphismFromCoDualToInternalCoHom as the identity of the CoDual" );
+  end
+] : CategoryFilter := IsRigidSymmetricCoclosedMonoidalCategory,
+    Description := "adding the isomorphisms between the internal cohom and the tensor product as identities"
```
Comment (zickgraf) [inline MonoidalCategories/gap/RigidSymmetricCoclosedMonoidalCategoriesDerivedMethods.gi:477]: see above

**Review by zickgraf — CHANGES_REQUESTED** [review]

File: MonoidalCategories/doc/Doc.autodoc [hunk 3]
```diff
@@ -90,6 +90,12 @@ A symmetric closed monoidal category $\mathbf{C}$ satisfying
  is an isomorphism
 is called a <Emph>rigid symmetric closed monoidal category</Emph>.
 
+The dual objects are given by $\mathrm{\underline{Hom}}(a,1) = a^\vee \otimes 1$
```
[the remark this answers is not in the document]
Comment (TKuh) [reply MonoidalCategories/doc/Doc.autodoc]: I added a similar comment for (co)closed monoidal.

File: MonoidalCategories/doc/Doc.autodoc [hunk 4]
```diff
@@ -58,6 +60,8 @@ which has for each functor $- \otimes b: \mathbf{C} \rightarrow \mathbf{C}$
 a left adjoint (denoted by $\mathrm{\underline{coHom}}(-,b)$)
 is called a <Emph>coclosed monoidal category</Emph>.
 
+If no operations involving coduals are installed manually, the codual objects will be derived as $\mathrm{\underline{coHom}}(1,a) = a_\vee$
```
Comment (zickgraf) [inline MonoidalCategories/doc/Doc.autodoc:63]:
```suggestion
If no operations involving coduals are installed manually, the codual objects will be derived as $a_\vee \coloneqq \mathrm{\underline{coHom}}(1,a)$.
```

File: MonoidalCategories/doc/Doc.autodoc [hunk 5]
```diff
@@ -90,6 +94,11 @@ A symmetric closed monoidal category $\mathbf{C}$ satisfying
  is an isomorphism
 is called a <Emph>rigid symmetric closed monoidal category</Emph>.
 
+If no operations involving duals are installed manually, the dual objects will be derived as $\mathrm{\underline{Hom}}(a,1) = a^\vee \otimes 1$
```
[the remark this answers is not in the document]
Comment (zickgraf) [reply MonoidalCategories/doc/Doc.autodoc:97]: I guess there are LaTeX-errors in the output of `make doc`? If yes, we have to add `\usepakage{mathtools}` in the `LateExtraPreamble` in `makedoc.g` (for an example see FinGSetsForCAP).

[the remark this answers is not in the document]
Comment (TKuh) [reply MonoidalCategories/doc/Doc.autodoc]: Thank you, that did the trick. :)

[the remark this answers is not in the document]
Comment (zickgraf) [reply MonoidalCategories/doc/Doc.autodoc]: Have you changed "\coloneqq" to ":=" deliberately?

[the remark this answers is not in the document]
Comment (TKuh) [reply MonoidalCategories/doc/Doc.autodoc]: I wanted to change it temporarily to ':=' to see how it looks and forgot to change it back... ugh. I'm sorry.

**Review by zickgraf — CHANGES_REQUESTED** [review]

File: MonoidalCategories/gap/RigidSymmetricClosedMonoidalCategoriesDerivedMethods.gi [hunk 6]
```diff
@@ -450,89 +451,30 @@ AddFinalDerivation( IsomorphismFromTensorProductToInternalHom,
...
-                      IsomorphismFromInternalHomToTensorProduct ],
-                    
+  end
+],
+[
+  IsomorphismFromInternalHomToDual,
   function( cat, object )
     
-    return IdentityMorphism( cat, DualOnObjects( cat, object ) );
+    return IdentityMorphism( cat, TensorProductOnObjects( cat, DualOnObjects( cat, object ), TensorUnit( cat ) ) );
```
[the remark this answers is not in the document]
Comment (TKuh) [reply MonoidalCategories/gap/RigidSymmetricClosedMonoidalCategoriesDerivedMethods.gi]: I think you are right. And having documented this is really helpful for correcting.

**Review by zickgraf — APPROVED** [review]
perfect, now also the diff in the compiled code is gone (up to the weights), which I think is a good sign :-)

**Pull request merged**